mitodl/teachersportal
# Added integration tests, refactored test utility code

#### What's this PR do?
- Refactors test code
- Adds some integration tests (see `CourseDetailPage_test.js`)
- Also changes behavior where a successful checkout will cause `buyTab` state will reset to initial state. Previously a successful checkout would leave the checkboxes there checked and the seat slider where it was before.
#### Where should the reviewer start?

`static/js/containers/CourseDetailPage_test.js`
#### How should this be manually tested?

For the UI change I mentioned, just go through a checkout and verify that the chapter checkboxes are unchecked after successful checkout.
#### What are the relevant tickets?

Fixes #200
### Any other relevant information?

The login dialog isn't tested because it's rendered outside the testing document. We might want to do this in a future PR.
#### What gif best describes this PR or how it makes you feel?

![](https://media.giphy.com/media/wUgWRubJHS7Ac/giphy.gif)
